### DISCUSSION TITLE
CDL- Cabinet-Description-Language - Config options before compile/finale releases

### DISCUSSION THREAD

**Discussion by WilsonBradley** [curif/AgeOfJoy-2022.1#110] [Ideas]
A place to put Config options before compile/finale releases

**Comment by WilsonBradley** [comment]
This is the config for Joust (goes in the joust cabinet folder)
The colors seem to be better with these settings.
[description.yaml.log](https://github.com/curif/AgeOfJoy-2022.1/files/10670455/description.yaml.log)
(Remove .log from file name) won't allow uploading of .yaml

**Comment by curif** [comment]
Global configuration and room configuration. Global stands for all rooms, and each room can have its own.

**Reply by curif** [reply]
done in #16

**Comment by curif** [comment]
Configuration in yaml files so power users can tweak the game from outside and share configs

**Reply by curif** [reply]
done in #16

**Comment by curif** [comment]
No screens, use world artifacts to configure the game. Buttons in control panels for example.

**Comment by curif** [comment]
Joysticks configuration to remap standard buttons.

**Comment by curif** [comment]
NPC config: disable, no arcade plays, stand static in a corner.

**Comment by curif** [comment]
CDL: change screen effect shaders. First new shader should be one that not change the libretro image.
Damage shader with high and medium damage.

**Comment by curif** [comment]
CDL: config dimensions for cabinets. Scale or shrink a cabinet.

**Comment by curif** [comment]
Background sound status: 
* disable
* volume

Maybe background sound alternatives to select.

**Comment by curif** [comment]
Eye level adjustment: to make players taller or shorter.
This parameter has no sense in each room, only globally.

**Comment by WilsonBradley** [comment]
Regards to the CDL scripting, lets make use of the # for commenting and notating what each line does and it's usable value range.
https://github.com/curif/AgeOfJoy-2022.1/wiki/CDL---Cabinet-Description-Language

**Reply by curif** [reply]
Maybe cabinet developers could sign off with comments like:

```
#author: curif
#license: xxx.xxx
```

**Reply by WilsonBradley** [reply]
That's a good idea to know who to reach out to.
I was also thinking for new users looking at/editing the file, would be helpful/useful to comment on each key entry and it's value range for better understanding. 
Once a 'Master' yaml file has been made with all the available keys known (we could just comment out if not currently used) 
Then all others could start updating older yaml / Cabinets to use the Master file you created as foundation, making it a standard.

What is licence ?

**Reply by curif** [reply]
License

**Reply by curif** [reply]
To have a Master yaml file it's a good idea as a standard foundation too.
